tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -0,0 +1,333 @@
+        public string TracesTransport { get; }
+
+        /// <summary>
+        /// Gets the windows pipe name where the Tracer can connect to the Agent.
+        /// Default is <c>null</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TracesPipeName"/>
+        public string TracesPipeName { get; }
+
+        /// <summary>
+        /// Gets the timeout in milliseconds for the windows named pipe requests.
+        /// Default is <c>100</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.TracesPipeTimeoutMs"/>
+        public int TracesPipeTimeoutMs { get; }
+
+        /// <summary>
+        /// Gets the windows pipe name where the Tracer can send stats.
+        /// Default is <c>null</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.MetricsPipeName"/>
+        public string MetricsPipeName { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether default Analytics are enabled.
+        /// Settings this value is a shortcut for setting
+        /// <see cref="Configuration.IntegrationSettings.AnalyticsEnabled"/> on some predetermined integrations.
+        /// See the documentation for more details.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.GlobalAnalyticsEnabled"/>
+        [Obsolete(DeprecationMessages.AppAnalytics)]
+        public bool AnalyticsEnabled { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether correlation identifiers are
+        /// automatically injected into the logging context.
+        /// Default is <c>false</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.LogsInjectionEnabled"/>
+        public bool LogsInjectionEnabled { get; }
+
+        /// <summary>
+        /// Gets a value indicating the maximum number of traces set to AutoKeep (p1) per second.
+        /// Default is <c>100</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.MaxTracesSubmittedPerSecond"/>
+        public int MaxTracesSubmittedPerSecond { get; }
+
+        /// <summary>
+        /// Gets a value indicating custom sampling rules.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.CustomSamplingRules"/>
+        public string CustomSamplingRules { get; }
+
+        /// <summary>
+        /// Gets a value indicating a global rate for sampling.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.GlobalSamplingRate"/>
+        public double? GlobalSamplingRate { get; }
+
+        /// <summary>
+        /// Gets a collection of <see cref="Integrations"/> keyed by integration name.
+        /// </summary>
+        public ImmutableIntegrationSettingsCollection Integrations { get; }
+
+        /// <summary>
+        /// Gets the global tags, which are applied to all <see cref="Span"/>s.
+        /// </summary>
+        public IDictionary<string, string> GlobalTags { get; }
```

Comment:
Why not IReadOnlyDictionary?

Reply:
Mostly because it meant I had to change a bunch of extension methods and interfaces, happy to do it though.

Reply:
Probably not really worth it, since the it's unlikely clients of this class will be tempted to use the add method.

Reply:
Since this is a public API (which will be harder to change later), I suggest we use `IReadOnlyDictionary` like Robert first suggested.

Reply:
Same for `HeaderTags` below.

Reply:
Yeah, makes sense. This does raise another question though: does this _need_ to be a public api? 🤔

Reply:
Replaced with `IReadOnlyDictionary` in https://github.com/DataDog/dd-trace-dotnet/pull/1951/commits/162f55e2748b1b19b2a3f4befae454e632dd93e3